src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -722,8 +722,16 @@ final case class AgreementApiServiceImpl(
     } yield ()
 
     for {
-      seed    <- getUpdateSeed()
+      seed               <- getUpdateSeed()
+      existingAgreements <- agreementManagementService.getAgreements(
+        consumerId = Some(agreement.consumerId.toString),
+        eserviceId = Some(agreement.eserviceId.toString)
+      )
+      archivables = extractArchivables(agreement.id, existingAgreements)
       updated <- agreementManagementService.updateAgreement(agreement.id, seed)
+      _       <-
+        if (archivables.isEmpty) Future.unit
+        else Future.traverse(archivables)(archive).map(_ => ())
```

Comment:
isn't this the same?
```suggestion
        Future.traverse(archivables)(archive).map(_ => ())
```